### PR TITLE
Add shape lowerings for a few missing operations.

### PR DESCRIPTION
**Pull request by rsuderman**

### CODE REVIEW

File: iree/compiler/Dialect/Shape/Conversion/ConvertShapeToShapex.cpp [hunk 1]
```diff
@@ -127,6 +127,46 @@ class ConvertShapeOfOp : public OpConversionPattern<shape::ShapeOfOp> {
...
+      }
+      dynOperands.push_back(operand);
+      extents.push_back(-1);
+    }
+
+    auto resultType = RankedShapeType::get(extents, rewriter.getContext());
+    auto make = rewriter.create<Shape::MakeRankedShapeOp>(
+        op.getLoc(), resultType, dynOperands);
+
+    make.dump();
```
Comment (silvasean) [inline iree/compiler/Dialect/Shape/Conversion/ConvertShapeToShapex.cpp:163]: remove dump

File: iree/compiler/Dialect/Shape/Conversion/test/shape_to_shapex.mlir [hunk 2]
```diff
@@ -45,6 +45,40 @@ func @f(%arg0: tensor<?xf32>, %index: i32) {
   return
 }
 
+// -----
+// shape.get_extent
+// CHECK-LABEL: func @f
+func @f(%arg0: tensor<?xf32>) {
+  %c0 = constant 0 : index
+  // CHECK: [[SHAPE:%.+]] = shapex.get_ranked_shape %arg0 : tensor<?xf32> -> !shapex.ranked_shape<[?]>
```
Comment (silvasean) [inline iree/compiler/Dialect/Shape/Conversion/test/shape_to_shapex.mlir:53]: Suggest `%[[SHAPE:.+]]` style for consistency.

File: iree/compiler/Dialect/Shape/Conversion/test/shape_to_shapex.mlir [hunk 3]
```diff
@@ -45,6 +45,40 @@ func @f(%arg0: tensor<?xf32>, %index: i32) {
...
+}
+
+// -----
+// shape.from_extents
+// CHECK-LABEL: func @f
+func @f(%arg0: tensor<?x?xf32>) {
+  %c0 = constant 0 : index
+  %c1 = constant 1 : index
+
+  // CHECK: [[SHAPE:%.]] = shapex.get_ranked_shape %arg0 : tensor<?x?xf32>
```
Comment (silvasean) [inline iree/compiler/Dialect/Shape/Conversion/test/shape_to_shapex.mlir:68]: Can we just test
```
func @f(%arg0: index, %arg1: index) {
   %result = "shape.from_extents"(%arg0, %arg1) : (index, index) -> !shape.shape
  return
}
```

?

**Review by silvasean — APPROVED** [review]

File: iree/compiler/Dialect/Shape/Conversion/test/shape_to_shapex.mlir (hunk 3 above)
Comment (rsuderman) [reply iree/compiler/Dialect/Shape/Conversion/test/shape_to_shapex.mlir]: Originally I wanted to check some of the intermediate types, moved to only testing with index.

File: iree/compiler/Dialect/Shape/Conversion/test/shape_to_shapex.mlir (hunk 2 above)
Comment (rsuderman) [reply iree/compiler/Dialect/Shape/Conversion/test/shape_to_shapex.mlir]: Done.

File: iree/compiler/Dialect/Shape/Conversion/ConvertShapeToShapex.cpp (hunk 1 above)
Comment (rsuderman) [reply iree/compiler/Dialect/Shape/Conversion/ConvertShapeToShapex.cpp]: Done.

**Pull request merged**